server/controllers/finance/reports/ohada_balance_sheet/index.js

```diff
@@ -0,0 +1,299 @@
+  { ref : 'BR', is_title : 0 },
+  { ref : 'BS', is_title : 0 },
+  { ref : 'BT', is_title : 1 },
+  { ref : 'BU', is_title : 0 },
+  { ref : 'BZ', is_title : 1 },
+];
+
+const balanceSheetLiabilityTable = [
+  { ref : 'CA', is_title : 1 },
+  { ref : 'CB', is_title : 0 },
+  { ref : 'CC', is_title : 0 },
+  { ref : 'CD', is_title : 0 },
+  { ref : 'CE', is_title : 0 },
+  { ref : 'CF', is_title : 0 },
+  { ref : 'CG', is_title : 0 },
+  { ref : 'CH', is_title : 0 },
+  { ref : 'CI', is_title : 1 },
+  { ref : 'CK', is_title : 1 },
+  { ref : 'CL', is_title : 0 },
+  { ref : 'CM', is_title : 0 },
+  { ref : 'CP', is_title : 1 },
+  { ref : 'DA', is_title : 0 },
+  { ref : 'DB', is_title : 0 },
+  { ref : 'DC', is_title : 0 },
+  { ref : 'DD', is_title : 0 },
+  { ref : 'DF', is_title : 1 },
+  { ref : 'DG', is_title : 1 },
+  { ref : 'DH', is_title : 0 },
+  { ref : 'DI', is_title : 0 },
+  { ref : 'DJ', is_title : 0 },
+  { ref : 'DK', is_title : 0 },
+  { ref : 'DM', is_title : 0 },
+  { ref : 'DN', is_title : 0 },
+  { ref : 'DP', is_title : 0 },
+  { ref : 'DQ', is_title : 0 },
+  { ref : 'DR', is_title : 0 },
+  { ref : 'DS', is_title : 0 },
+  { ref : 'DT', is_title : 1 },
+  { ref : 'DV', is_title : 1 },
+  { ref : 'DZ', is_title : 1 },
+];
+
+// expose to the API
+exports.document = document;
+
+/**
+ * @function document
+ * @description process and render the balance report document
+ */
+function document(req, res, next) {
+  const params = req.query;
+  const context = {};
+  let report;
+
+  _.defaults(params, DEFAULT_PARAMS);
+
+  try {
+    report = new ReportManager(TEMPLATE, req.session, params);
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+  getFiscalYearDetails(params.fiscal_id)
+    .then(fiscalYear => {
+      _.merge(context, { fiscalYear });
+
+      const dbPromises = [];
+      const usePrevious = fiscalYear.previous.period_id ?
+        AccountReference.computeAllAccountReference(fiscalYear.previous.period_id) : [];
+
+      dbPromises.push(AccountReference.computeAllAccountReference(fiscalYear.current.period_id));
+      dbPromises.push(usePrevious);
+      return Q.all(dbPromises);
+    })
+    .spread((current, previous) => {
+      let list = [];
+      const currentReferences = formatReferences(_.groupBy(current, 'abbr'));
+      const previousReferences = formatReferences(_.groupBy(previous, 'abbr'));
+
+      const assetTable = balanceSheetAssetTable.map(item => {
+        item.label = 'REPORT.OHADA.REF_DESCRIPTION.'.concat(item.ref);
+        if (currentReferences[item.ref]) {
+          item.currentBrut = currentReferences[item.ref].brut.balance;
+          item.currentAmo = currentReferences[item.ref].amortissement.balance;
+          item.currentNet = currentReferences[item.ref].net.balance;
+          item.previousNet = previousReferences[item.ref] ? previousReferences[item.ref].net.balance : 0;
+        }
+
+        // process manually totals
+        if (item.ref === 'AD') {
```

Comment:
👍 

Technically, you could do this all more efficiently with `switch`/`case` but this is really easy to read.  Good decision.